sockeye/inference.py

```diff
@@ -889,7 +889,10 @@ def __init__(self,
                  source_vocabs: List[vocab.Vocab],
                  target_vocab: vocab.Vocab,
                  restrict_lexicon: Optional[lexicon.TopKLexicon] = None,
-                 store_beam : bool = False) -> None:
+                 store_beam: bool = False,
+                 use_mxnet_topk: bool = False) -> None:
+        self.use_mxnet_topk = use_mxnet_topk
+        self.smallest_k_func = utils.smallest_k_mx if self.use_mxnet_topk else utils.smallest_k
```

Comment:
how about we branch based on `context`? Then we don't need the use_mxnet_topk flag at all (neither as a CLI argument nor in this constructor)